src/main.c

```diff
@@ -3583,25 +3586,26 @@ int sqlite3_open(
   sqlite3 **ppDb 
 ){
   return openDatabase(zFilename, ppDb,
-                      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL, NULL);
+                      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL, NULL, NULL);
 }
 int sqlite3_open_v2(
   const char *filename,   /* Database filename (UTF-8) */
   sqlite3 **ppDb,         /* OUT: SQLite db handle */
   int flags,              /* Flags */
   const char *zVfs        /* Name of VFS module to use */
 ){
-  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, NULL);
+  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, NULL, NULL);
 }
 
 int libsql_open(
   const char *filename,   /* Database filename (UTF-8) */
   sqlite3 **ppDb,         /* OUT: SQLite db handle */
   int flags,              /* Flags */
   const char *zVfs,       /* Name of VFS module to use, NULL for default */
-  const char *zWal        /* Name of WAL module to use */
+  const char *zWal,       /* Name of WAL module to use */
+  void* pMethodsData
```

Comment:
that's no good, we already released libSQL with this API, it's user-facing, and we should keep it backward compatible. Otherwise, existing applications that got compiled with the old interface would suddenly interpret as if this function has one more argument, and they would take garbage data from the stack as p*Wal*MethodsData. If we're about to change the API, we should introduce `libsql_open_v2`.

Also, while at it, we should probably document somewhere (docs/libsql_extensions.md?) that `libsql_wal_methods` are supposed to be stateless, and if you want to maintain state specific for wal methods, you should use `libsql_open_v2` and implement your wal methods so that they use this data by accessing it from `pWal->pMethods`.